packages/browser-wallet/src/popup/shared/TransactionReceipt/TransactionReceipt.tsx

```diff
@@ -76,3 +96,66 @@ export default function TransactionReceipt({
         </div>
     );
 }
+
+function TransactionReceipt({ transactionType, parameters, payload, ...props }: TransactionReceiptProps) {
+    return (
+        <TransactionReceiptView {...props} title={getTransactionTypeName(transactionType)}>
+            {displayPayload({ type: transactionType, payload }, parameters)}
+        </TransactionReceiptView>
+    );
+}
+
+export default function GenericTransactionReceipt(props: GenericTransactionReceiptProps) {
+    return <TransactionReceipt {...props} />;
+}
+
+export function TokenTransferReceipt({ parameters, symbol, metadata, ...props }: TokenTransferReceiptProps) {
+    const { t } = useTranslation('shared', { keyPrefix: 'transactionReceipt.tokenTransfer' });
+    const [advanced, setAdvanced] = useState(false);
+
+    const [
+        {
+            amount,
+            from: {
+                Account: [from],
+            },
+            to: {
+                Account: [to],
+            },
+        },
+    ] = parameters;
+
+    if (advanced) {
+        return (
+            <TransactionReceipt
+                {...props}
+                parameters={parameters}
+                onAlternate={() => setAdvanced(false)}
+                alternateText={t('showTransfer')}
+            />
+        );
+    }
+
+    return (
+        <TransactionReceiptView
+            {...props}
+            sender={from}
+            title={t('title', { symbol: trunctateSymbol(symbol) })}
```

Comment:
This looks weird for tokens without the symbol, would it not make more sense to show the name here?